store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedGraphStorage.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.federatedstore;
+
+import com.google.common.collect.Sets;
+
+import uk.gov.gchq.gaffer.graph.Graph;
+import uk.gov.gchq.gaffer.store.StoreTrait;
+import uk.gov.gchq.gaffer.store.exception.OverwritingException;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+import uk.gov.gchq.gaffer.store.schema.Schema.Builder;
+import uk.gov.gchq.gaffer.user.User;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class FederatedGraphStorage {
+    public static final String USER_IS_ATTEMPTING_TO_OVERWRITE_A_GRAPH_WITHIN_FEDERATED_STORE_GRAPH_ID_S = "User is attempting to overwrite a graph within FederatedStore. GraphId: %s";
+    public static final String CAN_NOT_PUT_GRAPH_INTO_STORAGE_WITHOUT_A_FEDERATED_ACCESS_KEY = "Can not put graph into storage without a FederatedAccess key.";
+    private Map<FederatedAccess, Set<Graph>> storage = new HashMap<>();
+    private Schema mergedSchema = new Schema();
+    private Set<StoreTrait> mergedTraits = Collections.emptySet();
+
+    public Collection<Graph> getAll(final User user) {
+        final HashSet<Graph> rtn = storage.entrySet()
+                .stream()
+                .filter(entry -> null != entry.getKey() && entry.getKey().isValidToExecute(user))
+                .flatMap(entry -> entry.getValue().stream())
+                .collect(Collectors.toCollection(HashSet::new));
+
+        return Collections.unmodifiableCollection(rtn);
+    }
+
+    public Collection<Graph> get(final User user, final Collection<String> graphIds) {
+        final HashSet<Graph> rtn = getAll(user)
+                .stream()
+                .filter(graph -> null == graphIds || graphIds.contains(graph.getGraphId()))
+                .collect(Collectors.toCollection(HashSet::new));
+
+        return Collections.unmodifiableSet(rtn);
+    }
+
+
+    public Collection<String> getAllIds(final User user) {
+        final HashSet<String> rtn = getAll(user)
```

Comment:
Similarly to the comment above, if possible it would be good to do this with a single hashset.

Reply:
👍